.github/workflows/upload.yml

```
@@ -0,0 +1,52 @@
+name: Upload release
+
+on:
+  # https://docs.github.com/en/webhooks/webhook-events-and-payloads#release
+  release:
+    types: [published]
+
+concurrency:
+  # Concurrency group that uses the workflow name and PR number if available
+  # or commit SHA as a fallback. If a new build is triggered under that
+  # concurrency group while a previous build is running it will be canceled.
+  # Repeated pushes to a PR will cancel all previous builds, while multiple
+  # merges to main will not cancel.
+  group: ${{ github.workflow }}-${{ github.ref_name || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  # create compressed tarballs and upload them to the published release
+  upload:
+    if: >-
+      !cancelled()
+      && !github.event.repository.fork
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        # test lower version (w/ defaults) and upper version (w/ defaults and conda-forge)
+        file-format: ['gz', 'bz2']
+    env:
+      FILE_NAME: "${{ github.event.repository.name }}-${{ github.ref_name }}.tar.${{ matrix.file-format }}"
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v4
+
+      - name: Gzip Source
+        if: matrix.file-format == 'gz'
+        run: tar -cfzv ${{ env.FILE_NAME }} ${{ github.workspace }}
```

this has the risk to compress things you don't want, this should be better (this can also run for both archive types instead of a matrix):
```suggestion
        run: git archive HEAD --output ${{ env.FILE_NAME }}
```

---

Oh you were faster with the review than expected, I've played with it a little more, I'm not sure what you may not want in the tarballs for the release artifacts?

---

reasons for not using tar:
* tar is not reproducible (depends on order in filesystem, locale, timezone, timestamps)
* if the workspace is dirty (e.g. when testing the command locally) then that dirty stuff gets also compressed (would require running `git clean -fdx && git reset --hard` before)... not an issue directly after checkout in that workflow

git archive was made just for the reason to create reproducible fs/os independent archives only compressing what exactly belongs to a commit.

---

Yeah, I knew about this, thanks for reminding me again